File: flapypan-blog-api/src/main/java/top/flapypan/blog/controller/LinkController.java
package top.flapypan.blog.controller;

import cn.dev33.satoken.annotation.SaIgnore;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;
import top.flapypan.blog.common.RestResult;
import top.flapypan.blog.entity.Link;
import top.flapypan.blog.service.LinkService;

import java.util.List;

@Slf4j
@RequiredArgsConstructor
@RestController
@RequestMapping("/link")
public class LinkController {

    private final LinkService linkService;

    @SaIgnore
    @GetMapping
    public RestResult<List<Link>> getAll() {
        return RestResult.success(linkService.getAll());
    }

    @PostMapping
    public RestResult<List<Link>> save(@RequestBody List<Link> links) {
        return RestResult.success(linkService.save(links));
    }

}

File: flapypan-blog-api/src/main/kotlin/top/flapypan/blog/controller/LinkController.kt
package top.flapypan.blog.controller

import org.springframework.web.bind.annotation.*
import top.flapypan.blog.common.restOk
import top.flapypan.blog.entity.Link
import top.flapypan.blog.service.LinkService

/**
 * 固定链接相关接口
 */
@RestController
@RequestMapping("/link")
class LinkController(
    private val linkService: LinkService
) {

    /**
     * 获取所有固定链接
     */
    @GetMapping
    fun all() = linkService.getAll().restOk()

    /**
     * 修改固定链接
     */
    @PostMapping
    fun save(@RequestBody links: List<Link>) = linkService.save(links).restOk()
}
